sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_data_lake_directory_client.py

```diff
@@ -104,14 +104,18 @@ def from_connection_string(
             - except in the case of AzureSasCredential, where the conflicting SAS tokens will raise a ValueError.
             If using an instance of AzureNamedKeyCredential, "name" should be the storage account name, and "key"
             should be the storage account key.
-        :paramtype credential: Optional[Union[str, Dict[str, str], "AzureNamedKeyCredential", "AzureSasCredential", "TokenCredential"]] = None,  # pylint: disable=line-too-long
+        :paramtype credential:
+            ~azure.core.credentials.AzureNamedKeyCredential or
+            ~azure.core.credentials.AzureSasCredential or
+            ~azure.core.credentials.TokenCredential or
+            str or dict[str, str] or None
```

Comment:
Oh, this looks nice! We've been struggling with a nice way to write this doc and have it render nicely. Assuming this renders nice, we'll apply this pattern everywhere we document credential.

Reply:
Yeah, we are using this pattern in our python packages. This is what it looks after rendering, it does renders nice :)
![image](https://github.com/Azure/azure-sdk-for-python/assets/20915761/5604e55b-b97c-4085-b20e-04a592019483)